pkg/provisioner/ironic/ironic.go

```diff
@@ -1130,6 +1130,12 @@ func (p *ironicProvisioner) Deprovision() (result provisioner.Result, err error)
 			nodes.ProvisionStateOpts{Target: nodes.TargetDeleted},
 		)
 
+	case nodes.CleanFail:
```

Comment:
`CleanFail` always results in maintenance mode, we have to unset it before we can do `manage`.

Reply:
I try to do `manage` verb from `cleanfail` without enter `maintenance` mode, and it worked smoothly.

Reply:
Right, but you'll be left with maintenance mode on.

Reply:
Ah ha, I have checked the workflow, and you was right. So should I unset `maintenance` flag after jump from `cleanfail` to `manage`?

```golang
	case nodes.CleanFail:
		result, err = p.changeNodeProvisionState(
			ironicNode,
			nodes.ProvisionStateOpts{Target: nodes.TargetManage},
		)
		if ironicNode.Maintenance {
			p.log.Info("unset host maintenance flag to make host ready again")
			p.setMaintenanceFlag(ironicNode, false)
		}
		return result, err
```

Reply:
I suggest unsetting maintenance before taking any actions.

Reply:
Thanks for your reviewing. The maintenance flag is unset before executing `manage` verb now.

Reply:
It's currently done *after* doing `manage`, which is not correct. Essentially no actions should be taken in maintenance.